src/applications/random.jl

```diff
@@ -0,0 +1,18 @@
+Random.gentype(::Type{<:Domain{T}}) where T = T
+
+# Random.gentype(::Type{<:ProductDomain{T}}) where T = T
+Base.rand(rng::AbstractRNG, s::Random.SamplerTrivial{<:ProductDomain}) = map(i->(rand(rng, i)), factors(s[]))
```

Comment:
I think here you'd want to put a `toexternalpoint(d, outcome_of_map)` around it, where `d` is the product domain